multiqc/modules/filtlong/filtlong.py

```diff
@@ -0,0 +1,114 @@
+from multiqc.modules.base_module import BaseMultiqcModule
+import logging
+from collections import OrderedDict
+from multiqc import config
+from multiqc.plots import bargraph
+
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="Filtlong",
+            anchor="filtlong",
+            href="https://github.com/rrwick/Filtlong",
+            info="A tool for filtering long reads by quality.",
+            doi="",
+        )
+
+        # Find and load reports
+        self.filtlong_data = dict()
+
+        # Find all files for filtlong
+        for f in self.find_log_files("filtlong", filehandles=True):
+            s_name = f["s_name"]
+            self.parse_logs(f)
+
+        self.filtlong_data = self.ignore_samples(self.filtlong_data)
+
+        if len(self.filtlong_data) == 0:
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.filtlong_data)))
+
+        # Write data to file
+        self.write_data_file(self.filtlong_data, "filtlong")
+        self.filtlong_general_stats()
+        self.target_bases_barplot()
+        self.keeping_bases_barplot()
+
+    def parse_logs(self, logfile):
+        """Parsing Logs. Note: careful of ANSI formatting log"""
+        file_content = logfile["f"]
+        for l in file_content:
+            # Find the valid metric
+            if "target:" in l:
+                s_name = logfile["s_name"]
+                self.add_data_source(logfile, s_name=s_name)
+                self.filtlong_data[s_name] = {}
+                self.filtlong_data[s_name]["Target bases"] = {}
+                self.filtlong_data[s_name]["Target bases"] = float(l.lstrip().split(" ")[1])
+
+            elif "keeping" in l:
+                self.filtlong_data[s_name]["Keeping bases"] = {}
+                self.filtlong_data[s_name]["Keeping bases"] = float(l.lstrip().split(" ")[1])
+
+            elif "fall below" in l:
+                self.filtlong_data[s_name]["Keeping bases"] = {}
+                self.filtlong_data[s_name]["Keeping bases"] = float(0)
```

Comment:
I'm a bit nervous about this. Does the log really mean that 0 bases are kept? I read it as meaning that _all_ bases were kept, rather than 0:

```
Filtering long reads
  target: 183248995 bp
  reads already fall below target after filtering

Outputting passed long reads
```

Now I'm not sure!

Reply:
Ok, I adjusted this in 66fea63ac86316ab56fc3802b23a346b0de90234 to simply omit this value instead of setting it to `0`. Hopefully that's safer.

We can review in the future if needed.